c7n/policy.py

```diff
@@ -628,6 +630,12 @@ def validate(self):
                     "resource:%s does not support cloudtrail mode policies" % (
                         self.policy.resource_type))
 
+    def run(self, event, lambda_context):
```

Comment:
Won't this delay before the event conditions are even evaluated?  I would think the conditions should be evaluated immediately and only delay before resources are queried.  FWIW, we added this to LambdaMode directly so we could insert the delay just before `resolve_resources()` is called.

Reply:
thats fair re event conditions first